packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -41,11 +56,27 @@ export class RedirectsMiddleware {
     return this.handler;
   }
 
+  protected excludeRoute(pathname: string) {
+    if (
+      pathname.includes('.') || // Ignore files
+      pathname.startsWith('/api/') || // Ignore Next.js API calls
+      pathname.startsWith('/sitecore/') || // Ignore Sitecore API calls
+      pathname.startsWith('/_next') // Ignore next service calls
+    ) {
+      return true;
+    }
+    return false;
+  }
+
   private handler = async (req: NextRequest): Promise<NextResponse> => {
     // Find the redirect from result of RedirectService
     const existsRedirect = await this.getExistsRedirect(req);
```

Comment:
We should check the `excludeRoute` before `existsRedirect` since the latter involves a fetch. Always want to do the cheap (performant) checks first.